torax/geometry.py

```diff
@@ -879,6 +880,213 @@ def from_fbt(
+            return area
+
+        eqfile = geometry_loader.load_geo_data(geometry_dir, geometry_file, geometry_loader.GeometrySource.EQDSK)
+
+        Rmaj = eqfile['xmag']
+        Rmin = eqfile['xbdry'].max() - Rmaj
+        B0   = eqfile['bcentre']
+
+        psi_eqdsk_1dgrid = np.linspace(eqfile['psimag'], eqfile['psibdry'], eqfile['nx'])
+
+        X_1D = np.linspace(eqfile['xgrid1'], eqfile['xgrid1'] + eqfile['xdim'], eqfile['nx'])
+        Z_1D = np.linspace(eqfile['zmid'] - eqfile['zdim']/2, eqfile['zmid'] + eqfile['zdim']/2, eqfile['nz'])
+        X, Z = np.meshgrid(X_1D, Z_1D, indexing='ij')
+        Xlcfs, Zlcfs = eqfile['xbdry'], eqfile['zbdry']
+
+        # Psi 2D grid defined on the Meshgrid
+        psi_eqdsk_2dgrid  = eqfile['psi']
+        # Create mask for the confined region, i.e.,Xlcfs.min() < X < Xlcfs.max(), Zlcfs.min() < Z < Zlcfs.max()
+
+        offset = 0.01
+        mask = (X > Xlcfs.min() - offset) & (X < Xlcfs.max() + offset) & (Z > Zlcfs.min() - offset) & (Z < Zlcfs.max() + offset)
+        masked_psi_eqdsk_2dgrid = np.ma.masked_where(~mask, psi_eqdsk_2dgrid)
+
+        # q on uniform grid (pressure, etc., also defined here)
+        q_eqdsk_1dgrid = eqfile['qpsi']
+
+        # ---- Interpolations
+        q_interp = scipy.interpolate.interp1d(psi_eqdsk_1dgrid, eqfile['qpsi'], kind='cubic')
+        psi_spline_fit   = scipy.interpolate.RectBivariateSpline(X_1D, Z_1D, psi_eqdsk_2dgrid, kx=3, ky=3, s=0)
+        F_interp = scipy.interpolate.interp1d(psi_eqdsk_1dgrid, eqfile['fpol'], kind='cubic') # toroidal field flux function
+
+        # -----------------------------------------------------------
+        # --------- Make flux surface contours ---------
+        # -----------------------------------------------------------
+
+        # set epsilon based on floating point precision of psi grid
+        # NOTE: this is hacky and should be replaced with a more robust method
+        epsilon = 1e-7
+
+        psi_interpolant  = np.linspace(eqfile['psimag'] + epsilon, eqfile['psibdry'] - epsilon, 100)
+
+        surfaces = []
+        cg_psi = contour_generator(X, Z, masked_psi_eqdsk_2dgrid)
+
+        for i, _psi in enumerate(psi_interpolant):
+            vertices = cg_psi.create_contour(_psi)
+            x_surface, z_surface = vertices[0].T[0], vertices[0].T[1]
+            surfaces.append((x_surface, z_surface))
+
+
+        # -----------------------------------------------------------
+        # --------- Compute Flux surface averages and 1D profiles ---------
+        # --- Area, Volume, R_inboard, R_outboard
+        # --- FSA: <1/R^2>, <Bp^2>, <|grad(psi)|>, <|grad(psi)|^2>
+        # --- Toroidal plasma current
+        # --- Integral dl/Bp
+        # -----------------------------------------------------------
+
+        # Gathering area for profiles
+        areas, volumes                = [], []
+        R_inboard, R_outboard         = [], []
+        flux_surf_avg_1_over_R2_eqdsk = [] # <1/R**2>
```

Comment:
nit: the underscore eqdsk is probably not necessary here?

To save vertical space etc is it worth preallocating the numpy arrays and then assigning in rather than needing to create the arrays and then turn them into arrays?

Reply:
I think the underscore is useful as the `from_chease` method has similar syntax for the FSA quantities.